### PR TITLE
Enable Kondo for tests part 2: enable `:unused-binding` linter and fix warnings

### PR DESCRIPTION
**Pull request by camsaul**
Part of #24727
Follow-on to #24736

This PR enables the clj-kondo `:unused-binding` linter for tests and fixes ~350 warnings. 

I was a little unsure whether this linter had real value for our test code but after fixing all the warnings I have concluded that it in fact does. Almost all of the warnings were trivial like destructuring a key that didn't end up getting used or naming an object created with `with-temp` but not using that object directly, but there were a handful of warnings that exposed real errors in the test code, or places where we did unnecessary app DB calls for stuff we didn't end up using. I will call out a few examples in comments in this PR

### CODE REVIEW

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/models/permissions/block_permissions_test.clj [hunk 1]
```diff
@@ -226,7 +226,7 @@
                   (doseq [[message perms] {"with full DB perms"                   (perms/data-perms-path (mt/id))
                                            "with perms for the Table in question" (perms/table-query-path (mt/id :venues))}]
                     (mt/with-temp Permissions [_ {:group_id group-2-id, :object perms}]
-                      (testing "Should be able to run the query"
+                      (testing (format "Should be able to run the query %s" message)
```
Comment (camsaul) [inline enterprise/backend/test/metabase_enterprise/advanced_permissions/models/permissions/block_permissions_test.clj:229]: Here's one example: `message` above was unused but it was clearly meant to be part of the `testing` context

File: enterprise/backend/test/metabase_enterprise/audit_app/pages/common_test.clj [hunk 2]
```diff
@@ -20,12 +19,11 @@
 
 (defmethod audit.i/internal-query ::legacy-format-query-fn
   [_ a1]
-  (let [h2? (= (mdb/db-type) :h2)]
```
Comment (camsaul) [inline enterprise/backend/test/metabase_enterprise/audit_app/pages/common_test.clj:23]: `h2?` here was unused so removing it actually let us eliminate a namespace dependency

**Comment by dpsutton** [comment]
> /work/test/metabase/query_processor_test/expressions_test.clj:339:9: warning: unused binding dataset-def
from the test failure.

Godo addition. Makes for a tidy codebase.

File: test/metabase/pulse/render/body_test.clj [hunk 3]
```diff
@@ -648,7 +643,7 @@
 
 (def donut-info #'body/donut-info)
 
-(deftest donut-info-test
+(deftest ^:parallel donut-info-test
```
Comment (camsaul) [inline test/metabase/pulse/render/body_test.clj:646]: While I was going thru everything I was able to make a few more tests `:^parallel`. There are probably more in this namespace but I didn't check everything

**Review by dpsutton — APPROVED** [review]

File: test/metabase/models/collection_test.clj [hunk 4]
```diff
@@ -1461,25 +1460,22 @@
...
-      (mt/with-temp* [Card       [{card-id :id}]
-                      Collection [{collection-id :id} {:namespace "x"}]]
+      (mt/with-temp Collection [{collection-id :id} {:namespace "x"}]
         (is (thrown-with-msg?
              clojure.lang.ExceptionInfo
              #"A Card can only go in Collections in the \"default\" namespace"
              (collection/check-collection-namespace Card collection-id)))))
 
     (testing "Should throw exception if Collection does not exist"
-      (mt/with-temp Card [{card-id :id}]
```
Comment (camsaul) [inline test/metabase/models/collection_test.clj:1478]: Creating stuff not even used in or needed by the tests. The linter caught that. So this PR will actually make our tests just a little bit faster.

File: test/metabase/query_processor/middleware/add_dimension_projections_test.clj [hunk 5]
```diff
@@ -355,11 +355,7 @@
                                     :id           %categories.name
                                     :display_name "Category → Name"
                                     :fk_field_id  %category_id
-                                    :options      {::qp.add-dimension-projections/new-field-dimension-id dimension-id}}]})
-                rows     [["20th Century Cafe"               2 "Café"]
-                          ["25°"                             2 "Burger"]
-                          ["33 Taps"                         2 "Bar"]
-                          ["800 Degrees Neapolitan Pizzeria" 2 "Pizza"]]]
```
Comment (camsaul) [inline test/metabase/query_processor/middleware/add_dimension_projections_test.clj:362]: more noise that wasn't even used in the tests

File: test/metabase/query_processor/middleware/fetch_source_query_test.clj [hunk 6]
```diff
@@ -75,21 +75,18 @@
                                  "2015-02-01"]})))))))
   (testing "respects `enable-nested-queries` server setting"
     (mt/with-temp* [Card [{card-id :id} {:dataset_query (mt/mbql-query venues)}]]
-      (letfn [(resolve [] (resolve-card-id-source-tables
-                           (mt/mbql-query nil
-                                          {:source-table (str "card__" card-id)})))]
```
Comment (camsaul) [inline test/metabase/query_processor/middleware/fetch_source_query_test.clj:80]: unused function

File: test/metabase/db/schema_migrations_test.clj [hunk 7]
```diff
@@ -69,7 +69,7 @@
         (migrate!)
         (doseq [e [e1 e2]]
           (is (= true
-                 (db/exists? User :email (u/lower-case-en e1)))))))))
```
Comment (camsaul) [inline test/metabase/db/schema_migrations_test.clj:72]: this is probably the worst thing I found -- it was supposed to be checking a condition for both values of `e` (`e1` and `e2`) but instead it actually ended up checking `e1` twice and not checking `e2` at all

**Pull request merged**